plugin/src/main/groovy/com/novoda/gradle/command/Install.groovy

```diff
@@ -4,29 +4,47 @@ import org.gradle.api.tasks.TaskAction
 
 class Install extends AdbTask {
 
-    def customFlags
+    InstallExtension installExtension
+
+    /**
+     * Manual creation of Install task is deprecated.
+     * Please refer to install dsl documentation for details:
+     * https://github.com/novoda/gradle-android-command-plugin#install
+     */
+    @Deprecated
+    void setCustomFlags(customFlags) {
+        logger.warn """\
+                       Manual creation of Install task is deprecated.
+                       Please refer to install dsl documentation and modify your task '$name'
+                       https://github.com/novoda/gradle-android-command-plugin#install
+                       """.stripIndent()
+        installExtension = new InstallExtension()
+        installExtension.customFlags = customFlags
+    }
 
     @TaskAction
     void exec() {
         def arguments = ['install']
-        if (getCustomFlags())
-            arguments += getCustomFlags()
+        if (resolveCustomFlags())
+            arguments += resolveCustomFlags()
 
-        arguments += ['-r', '-d', apkPath]
+        arguments += ['-r', apkPath]
 
         assertDeviceAndRunCommand(arguments)
     }
 
-    private getCustomFlags() {
-        if (customFlags instanceof Closure)
-            customFlags = customFlags.call()
-        customFlags ?: ''
+    private resolveCustomFlags() {
+        def customFlags = installExtension.customFlags
+        if (customFlags instanceof Closure) {
+            return customFlags.call() ?: ''
+        }
+        return customFlags ?: ''
     }
 
     @Override
     protected handleCommandOutput(def text) {
         super.handleCommandOutput(text)
-        def matcher = text =~ /Failure \[(.*?)\]/
+        def matcher = text =~ /Failure \[(.*?)]/
```

Comment:
Was this broken?

Reply:
Nope. It was a suggestion from IntelliJ. Appearantly you don't need to escape that. Both works. IntelliJ was saying that it is redundant.